aptos-move/move-examples/dao/nft_dao/sources/nft_dao.move

```diff
@@ -0,0 +1,609 @@
+            // this token is a member token
+            if (vote) {
+                stats.total_yes = stats.total_yes + 1;
+                bucket_table::add(&mut stats.yes_votes, token_id, voter_addr);
+            } else {
+                stats.total_no = stats.total_no + 1;
+                bucket_table::add(&mut stats.no_votes, token_id, voter_addr);
+            };
+            i = i + 1;
+        };
+    }
+
+    /// Entry function that can be called by anyone
+    public entry fun resolve(proposal_id: u64, nft_dao: address) acquires Proposals, DAO, ProposalVotingStatistics {
+        // validate if proposal is ready to resolve
+        let dao = borrow_global_mut<DAO>(nft_dao);
+
+        // assert the proposal voting ended
+        let proposals = borrow_global_mut<Proposals>(nft_dao);
+        let proposal = table::borrow_mut(&mut proposals.proposals, proposal_id);
+        let now = timestamp::now_seconds();
+        assert!(now >= proposal.start_time_sec + dao.voting_duration, error::invalid_argument(EPROPOSAL_ENDED));
+
+        // assert the proposal is unresolved yet
+        assert!(proposal.resolution == PROPOSAL_PENDING, error::invalid_argument(EPROPOSAL_RESOLVED));
+
+        let proposal_stat = &mut borrow_global_mut<ProposalVotingStatistics>(nft_dao).proposals;
+        let voting_stat = table::borrow_mut(proposal_stat, proposal_id);
+        proposal.final_yes_vots = voting_stat.total_yes;
+        proposal.final_no_votes = voting_stat.total_no;
+        // validate resolve threshold and result
+        let voted = voting_stat.total_no + voting_stat.total_yes;
+        if (voted < dao.resolve_threshold) {
+            // not sufficient token voted
+            proposal.resolution = PROPOSAL_RESOLVED_NOT_PASSED;
+        } else if(voting_stat.total_yes > voting_stat.total_no) {
+            let function_name = proposal.function_name;
+            if (function_name == string::utf8(b"transfer_fund")) {
+                // This is very dangerous.
+                // We should exploring have server side dynamic compiling and deploying DAO contracts in each DAO's own accounts.
+                let res_signer =  create_signer_with_capability(&dao.dao_signer_capability);
+                let dst_addr = property_map::read_address(&proposal.function_args, &string::utf8(b"dst"));
+                let amount = property_map::read_u64(&proposal.function_args, &string::utf8(b"amount"));
+                transfer_fund(&res_signer, dst_addr, amount);
+            } else {
+               assert!(function_name == string::utf8(b"no_op"), error::invalid_argument(ENOT_SUPPROTED_FUNCTION));
+            };
+            proposal.resolution = PROPOSAL_RESOLVED_PASSED;
+        } else {
+            proposal.resolution = PROPOSAL_RESOLVED_NOT_PASSED;
+        };
+    }
+
+    /// DAO admin can transfer its DAO to another admin account
```

Comment:
I think the two step process is 
1. generally good for preventing the receiver getting the undesired NFTs
2. this won't help avoid admin authority being transferred to the incorrect address right? 

Happy to create a follow-up issue/PR for the admin transferring and continue the discussion there. 
This PR is already long and we can focus on getting existing features ready. I can remove the current transfer_admin function to avoid causing confusion as well.